pkg/engine/templates_generated.go

```diff
@@ -13357,6 +13401,21 @@ source {{GetCSEHelpersScriptFilepath}}
 wait_for_file 3600 1 {{GetCSEInstallScriptFilepath}} || exit $ERR_FILE_WATCH_TIMEOUT
 source {{GetCSEInstallScriptFilepath}}
 
+{{- if HasTelemetryEnabled }}
+installAPMZ
+
+function exitAndFlush {
+  if [[ $? != "0" ]]; then
+    trace_err "cse-exit" code=$?
+  else
+    trace_info "cse-exit" code=$?
+  fi
+  apmz batch -f ${TMP_APMZ_BATCH_FILE} --api-key {{GetApplicationInsightsTelemetryKey}}
+}
+
+trap exitAndFlush EXIT
```

Comment:
this is 💯 

I meant to do something like this a while ago and actually didn't know you could do it like this, just learned something